molgenis/client.py

```diff
@@ -42,21 +48,24 @@ def login(self, username, password):
         response = self._session.post(self._url + "v1/login",
                                       data=json.dumps({"username": username, "password": password}),
                                       headers={"Content-Type": "application/json"})
-        if response.status_code == 200:
-            self._token = response.json()["token"]
+        try:
+            response.raise_for_status()
+        except Exception as ex:
```

Comment:
On second thought:

I think you should catch `HttpError` here, because those are the only kind of exceptions that contain error messages from Molgenis.

Reply:
I started using the HttpError, but that was not sufficient. I think the RequestsException is a nice suggestion.